armory/datasets/preprocessing.py

```diff
@@ -58,6 +58,47 @@ def carla_over_obj_det_dev(element, modality="rgb"):
     )
 
 
+def carla_video_tracking_preprocess(x, max_frames=None):
+    # Clip
+    if max_frames:
+        max_frames = int(max_frames)
+        if max_frames <= 0:
+            raise ValueError(f"max_frames {max_frames} must be > 0")
+        x = x[:max_frames, :]
+    return x
+
+
+def carla_video_tracking_preprocess_labels(y, y_patch_metadata, max_frames=None):
+    # Clip
+    if max_frames:
+        max_frames = int(max_frames)
+        if max_frames <= 0:
+            raise ValueError(f"max_frames {max_frames} must be > 0")
+        y = y[:max_frames, :]
+        y_patch_metadata = {k: v[:max_frames, :] for (k, v) in y_patch_metadata.items()}
+    # Update labels
+    box_array = tf.squeeze(y, axis=0) if y.shape[0] == 1 else y
```

Comment:
are we ever expecting the squeeze condition `y.shape[0] == 1` to be True? To my knowledge, no, and if this is the case we should remove these lines as it only would then be adding potential confusion

Reply:
Addressed in a82f34fcfaf1c03e55a14257fa98452790a8ccda